### ISSUE TITLE
Failed to get database name and table name

### ISSUE THREAD

**Issue by jiangxiluning** [confluentinc/kafka-connect-hdfs#201]
How to reproduce:
It seems worker thread failed to create new table in the matastore.

1. delete hdfsconnector connector
2. create hdfsconnector connector
3. produce some avro messages

```
[2017-05-31 19:41:14,822] INFO WorkerSinkTask{id=test-hdfs-sink-0} Committing offsets (org.apache.kafka.connect.runtime.WorkerSinkTask:272)
[2017-05-31 19:41:14,823] WARN Commit of WorkerSinkTask{id=test-hdfs-sink-0} offsets timed out (org.apache.kafka.connect.runtime.WorkerSinkTask:172)
[2017-05-31 19:41:23,351] ERROR Task test-hdfs-sink-0 threw an uncaught and unrecoverable exception (org.apache.kafka.connect.runtime.WorkerSinkTask:449)
java.lang.RuntimeException: java.util.concurrent.ExecutionException: io.confluent.connect.hdfs.errors.HiveMetaStoreException: Invalid table
	at io.confluent.connect.hdfs.DataWriter.write(DataWriter.java:226)
	at io.confluent.connect.hdfs.HdfsSinkTask.put(HdfsSinkTask.java:103)
	at org.apache.kafka.connect.runtime.WorkerSinkTask.deliverMessages(WorkerSinkTask.java:429)
	at org.apache.kafka.connect.runtime.WorkerSinkTask.poll(WorkerSinkTask.java:250)
	at org.apache.kafka.connect.runtime.WorkerSinkTask.iteration(WorkerSinkTask.java:179)
	at org.apache.kafka.connect.runtime.WorkerSinkTask.execute(WorkerSinkTask.java:148)
	at org.apache.kafka.connect.runtime.WorkerTask.doRun(WorkerTask.java:139)
	at org.apache.kafka.connect.runtime.WorkerTask.run(WorkerTask.java:182)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.util.concurrent.ExecutionException: io.confluent.connect.hdfs.errors.HiveMetaStoreException: Invalid table
	at java.util.concurrent.FutureTask.report(FutureTask.java:122)
	at java.util.concurrent.FutureTask.get(FutureTask.java:192)
	at io.confluent.connect.hdfs.DataWriter.write(DataWriter.java:220)
	... 12 more
Caused by: io.confluent.connect.hdfs.errors.HiveMetaStoreException: Invalid table
	at io.confluent.connect.hdfs.hive.HiveMetaStore.createTable(HiveMetaStore.java:200)
	at io.confluent.connect.hdfs.parquet.ParquetHiveUtil.createTable(ParquetHiveUtil.java:46)
	at io.confluent.connect.hdfs.TopicPartitionWriter$1.call(TopicPartitionWriter.java:640)
	at io.confluent.connect.hdfs.TopicPartitionWriter$1.call(TopicPartitionWriter.java:637)
	... 4 more
Caused by: InvalidObjectException(message:t-im-message is not a valid object name)
	at org.apache.hadoop.hive.metastore.api.ThriftHiveMetastore$create_table_with_environment_context_result$create_table_with_environment_context_resultStandardScheme.read(ThriftHiveMetastore.java:29974)
	at org.apache.hadoop.hive.metastore.api.ThriftHiveMetastore$create_table_with_environment_context_result$create_table_with_environment_context_resultStandardScheme.read(ThriftHiveMetastore.java:29951)
	at org.apache.hadoop.hive.metastore.api.ThriftHiveMetastore$create_table_with_environment_context_result.read(ThriftHiveMetastore.java:29877)
	at org.apache.thrift.TServiceClient.receiveBase(TServiceClient.java:78)
	at org.apache.hadoop.hive.metastore.api.ThriftHiveMetastore$Client.recv_create_table_with_environment_context(ThriftHiveMetastore.java:1075)
	at org.apache.hadoop.hive.metastore.api.ThriftHiveMetastore$Client.create_table_with_environment_context(ThriftHiveMetastore.java:1061)
	at org.apache.hadoop.hive.metastore.HiveMetaStoreClient.create_table_with_environment_context(HiveMetaStoreClient.java:2050)
	at org.apache.hadoop.hive.metastore.HiveMetaStoreClient.createTable(HiveMetaStoreClient.java:669)
	at org.apache.hadoop.hive.metastore.HiveMetaStoreClient.createTable(HiveMetaStoreClient.java:657)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.hadoop.hive.metastore.RetryingMetaStoreClient.invoke(RetryingMetaStoreClient.java:152)
	at com.sun.proxy.$Proxy52.createTable(Unknown Source)
	at io.confluent.connect.hdfs.hive.HiveMetaStore$5.call(HiveMetaStore.java:185)
	at io.confluent.connect.hdfs.hive.HiveMetaStore$5.call(HiveMetaStore.java:182)
	at io.confluent.connect.hdfs.hive.HiveMetaStore.doAction(HiveMetaStore.java:87)
	at io.confluent.connect.hdfs.hive.HiveMetaStore.createTable(HiveMetaStore.java:193)
	... 7 more
[2017-05-31 19:41:23,359] ERROR Task is being killed and will not recover until manually restarted (org.apache.kafka.connect.runtime.WorkerSinkTask:450)
[2017-05-31 19:41:23,397] ERROR Task test-hdfs-sink-0 threw an uncaught and unrecoverable exception (org.apache.kafka.connect.runtime.WorkerTask:141)
org.apache.kafka.connect.errors.ConnectException: Exiting WorkerSinkTask due to unrecoverable exception.
	at org.apache.kafka.connect.runtime.WorkerSinkTask.deliverMessages(WorkerSinkTask.java:451)
	at org.apache.kafka.connect.runtime.WorkerSinkTask.poll(WorkerSinkTask.java:250)
	at org.apache.kafka.connect.runtime.WorkerSinkTask.iteration(WorkerSinkTask.java:179)
	at org.apache.kafka.connect.runtime.WorkerSinkTask.execute(WorkerSinkTask.java:148)
	at org.apache.kafka.connect.runtime.WorkerTask.doRun(WorkerTask.java:139)
	at org.apache.kafka.connect.runtime.WorkerTask.run(WorkerTask.java:182)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
[2017-05-31 19:41:23,398] ERROR Task is being killed and will not recover until manually restarted (org.apache.kafka.connect.runtime.WorkerTask:142)
[2017-05-31 19:41:23,399] INFO Shutting down Hive executor service. (io.confluent.connect.hdfs.DataWriter:309)
[2017-05-31 19:41:23,399] INFO Awaiting termination. (io.confluent.connect.hdfs.DataWriter:312)
```

Below is my connector setting and worker setting:
```
{
    "connector.class":"io.confluent.connect.hdfs.HdfsSinkConnector",
    "topics":"t-im-message",
    "tasks.max":1,
    "hdfs.url":"hdfs://goome",
    "flush.size":1,
    "hadoop.conf.dir":"/etc/hadoop/conf",
    "format.class":"io.confluent.connect.hdfs.parquet.ParquetFormat",
    "hive.integration":true,
    "hive.metastore.uris":"thift://da100:9083",
    "hive.conf.dir":"/etc/hive/conf",
    "schema.compatibility":"BACKWARD",
    "hive.database":"community_logs"
}
```
```
# Bootstrap Kafka servers. If multiple servers are specified, they should be comma-separated.
bootstrap.servers=da100:9092,da103:9092,da106:9092

# The group ID is a unique identifier for the set of workers that form a single Kafka Connect
# cluster
group.id=connect-cluster

# The converters specify the format of data in Kafka and how to translate it into Connect data.
# Every Connect user will need to configure these based on the format they want their data in
# when loaded from or stored into Kafka

key.converter=io.confluent.connect.avro.AvroConverter
key.converter.schema.registry.url=http://da100:8083
value.converter=io.confluent.connect.avro.AvroConverter
value.converter.schema.registry.url=http://da100:8083

# The internal converter used for offsets and config data is configurable and must be specified,
# but most users will always want to use the built-in default. Offset and config data is never
# visible outside of Connect in this format.
internal.key.converter=org.apache.kafka.connect.json.JsonConverter
internal.value.converter=org.apache.kafka.connect.json.JsonConverter
internal.key.converter.schemas.enable=false
internal.value.converter.schemas.enable=false

# Kafka topic where connector configuration will be persisted. You should create this topic with a
# single partition and high replication factor (e.g. 3)
config.storage.topic=connect-configs

# Kafka topic where connector offset data will be persisted. You should create this topic with many
# partitions (e.g. 25) and high replication factor (e.g. 3)
offset.storage.topic=connect-offsets

# Kafka topic where connector status data will be persisted. You should create this topic with many
# partitions (e.g. 25) and high replication factor (e.g. 3)
status.storage.topic=connect-status

rest.port=18084

offset.flush.interval.ms=10000
heartbeat.interval.ms=5000
```

**Comment by kkonstantine** [comment]
This seems to be a known issue with hive table names not supporting hyphens. We've added a patch to storage-common (https://github.com/confluentinc/kafka-connect-storage-common/pull/24) as part of the general hdfs connector refactoring, but we should probably backport this fix here too.

**Comment by cricket007** [comment]
@kkonstantine This can be closed with the 4.0 refactoring, yes?

**Comment by levzem** [comment]
closing as this is fixed now
